platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java

```diff
@@ -168,6 +173,28 @@
         return hashedData.getDescriptor();
     }
 
+    @Override
+    public Iterator<Transaction> transactionIterator() {
+        return Arrays.asList((Transaction[]) hashedData.getTransactions()).iterator();
```

Comment:
Do you need to check if `hashedData.getTransactions()` returns `null`?

Reply:
In practice, transaction should never be null, although the legacy code pathways allow it to happen in theory. This theoretical pathway should be closed soon.